### PR TITLE
[FSSDK-10950] refactor ODP implementation

### PR DESCRIPTION
**Pull request by raju-opti**
## Summary
- Refactored odp implementaion

## Issues
- FSSDK-10950

### CODE REVIEW

File: lib/vuid/vuid_manager_factory.browser.spec.ts [hunk 1]
```diff
@@ -0,0 +1,68 @@
+import { vi, describe, expect, it, beforeEach } from 'vitest';
```
Comment (junaed-optimizely) [inline lib/vuid/vuid_manager_factory.browser.spec.ts:1]: Please add copyright text here!

File: lib/vuid/vuid_manager_factory.node.spec.ts
Comment (junaed-optimizely) [inline lib/vuid/vuid_manager_factory.node.spec.ts:1]: Please add copyright text!

File: lib/vuid/vuid_manager_factory.ts
Comment (junaed-optimizely) [inline lib/vuid/vuid_manager_factory.ts:1]: Please add copyright text!

File: lib/odp/event_manager/odp_event_api_manager.spec.ts [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+ */
+
+import { describe, beforeEach, beforeAll, it, expect, vi } from 'vitest';
+
+import { LogHandler, LogLevel } from '../../modules/logging';
+import { DefaultOdpEventApiManager, eventApiRequestGenerator, pixelApiRequestGenerator } from './odp_event_api_manager';
+import { OdpEvent } from './odp_event';
+import { RequestHandler } from '../../utils/http_request_handler/http';
+import { OdpConfig } from '../odp_config';
+import { get } from 'http';
```
Comment (junaed-optimizely) [inline lib/odp/event_manager/odp_event_api_manager.spec.ts:24]: NIT: This whole import block has some unused imports. It could use some cleanups

**Review by junaed-optimizely — APPROVED** [review]
LGTM!

All most all of the issues are regarding copyright text, and some minor cleanups!

File: lib/index.browser.tests.js
[the remark this answers is not in the document]
Comment (raju-opti) [reply lib/index.browser.tests.js:1]: Yeah, this file will be completely rewritten soon, it contained a lot of tests that belong in other places. keeping it as it is for now.

File: lib/odp/event_manager/odp_event_manager.ts [hunk 3]
```diff
@@ -14,440 +14,198 @@
...
-  }
 
-  /**
-   * Clear the currently running timout
-   * @private
-   */
-  private clearCurrentTimeout(): void {
-    clearTimeout(this.timeoutId);
-    this.timeoutId = undefined;
+    this.flush();
```
[the remark this answers is not in the document]
Comment (raju-opti) [reply lib/odp/event_manager/odp_event_manager.ts:145]: It was not done in the previous implementation, so I thought maybe keep it a bit simpler and don't do here either. 

We probably should do this and add tests in a later cleanup stage

**Pull request merged**